test/convergence/test_mini_models.py

```diff
@@ -331,6 +331,7 @@ def run_mini_model(
 
     for i in range(num_steps):
         batch = next(loader_iter).to(model.device)
+        optimizer.zero_grad()
```

Comment:
nit: should this be after line 337?

Reply:
Anywhere in the loop that's not between backwards() and step() is valid - PyTorch tutorials tend to place it at the beginning of a training step.